lib/project_types/script/layers/infrastructure/languages/assemblyscript_project_creator.rb

```diff
@@ -5,14 +5,29 @@ module Layers
     module Infrastructure
       module Languages
         class AssemblyScriptProjectCreator < ProjectCreator
-          def self.config_file
-            "package.json"
-          end
-
           def setup_dependencies
             task_runner = Infrastructure::Languages::AssemblyScriptTaskRunner.new(ctx)
             task_runner.set_npm_config
             super
+
+            update_package_json_name
+          end
+
+          private
+
+          def update_package_json_name
+            file_content = ctx.read("package.json")
+            hash = file_content_to_hash(file_content)
+            hash["name"] = project_name
+            ctx.write("package.json", hash_to_file_content(hash))
+          end
+
+          def file_content_to_hash(content)
+            JSON.parse(content)
+          end
+
+          def hash_to_file_content(hash)
+            JSON.pretty_generate(hash)
           end
```

Comment:
My :tophat: suggests the order of keys is maintained with this, which is surprising to me but really convenient. I was going to suggest using `file_content_to_hash(file_content)["name"]` to get the name and then use `gsub` on `file_content` to preserve the order of keys, but looks like that is not necessary.